### PR TITLE
Make responses no-store by default

### PR DESCRIPTION
**Pull request by tvlooy**
| Q             | A
| ------------- | ---
| Branch?       | 6.2 for features
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| Tickets       | 
| License       | MIT
| Doc PR        | 

The AbstractSessionListener already makes the responses max-age=0 and private. That means that intermediate caches won't store the response and private caches won't use it. But, browser caches still store the response.

Adding the no-store will fix that. See rfc7234 5.2.2.3: "the cache MUST NOT intentionally store the information in non-volatile storage, and MUST make a best-effort attempt to remove the information from volatile storage as promptly as possible after forwarding it"

When returning sensitive information from your application, this makes it a little less likely that that info can somehow be retrieved from browser caches.

We already have a max-age=0 and private, so this seems to be a good addition to make it a little bit more secure by default.

### CODE REVIEW

File: src/Symfony/Component/HttpKernel/EventListener/AbstractSessionListener.php [hunk 1]
```diff
@@ -197,6 +197,7 @@ public function onKernelResponse(ResponseEvent $event)
         if ($autoCacheControl) {
             $response
                 ->setExpires(new \DateTimeImmutable())
+                ->setNoStore()
```
Comment (alexander-schranz) [inline src/Symfony/Component/HttpKernel/EventListener/AbstractSessionListener.php:200]: I personally would not do this by default. I mean its fine in most cases even when having a session that History Back of the browser does not make any request to the server and just load the content from the browser history and don't use server resources.

Still think the method make sense and can for this who need it be called in the `index.php` to make all there responses noStore.

**Review by nicolas-grekas — COMMENTED** [review]
I'm :-1: on this one. This is not required to make the session tracking logic correct and this would break the client-side cache.

**Comment by tvlooy** [comment]
@nicolas-grekas client-side cache as in "the history back button" then? So, if you do want this behavior on your project the best solution is to add your own session handler that adds no-store on the responses?

**Comment by nicolas-grekas** [comment]
Or a response listener really.

**Comment by tvlooy** [comment]
ah yes, response listener, right :+1:

**Pull request closed without merging**